diff --git a/toolchain/args/BUILD.bazel b/toolchain/args/BUILD.bazel
index 0000000..0000000 100644
--- a/toolchain/args/BUILD.bazel
+++ b/toolchain/args/BUILD.bazel
@@ -115,7 +115,7 @@ cc_args(
         "@rules_cc//cc/toolchains/actions:lto_index_for_executable",
         "@rules_cc//cc/toolchains/actions:lto_index_for_dynamic_library",
         "@rules_cc//cc/toolchains/actions:lto_index_for_nodeps_dynamic_library",
     ],
     args = [
-        "-static-pie",
+        "-static",
     ],
 )
