diff --git a/repositories/github.go b/repositories/github.go
index 0d02067..24b12e7 100644
--- a/repositories/github.go
+++ b/repositories/github.go
@@ -3,6 +3,7 @@ package repositories
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/bazelbuild/bazelisk/httputil"
 	"github.com/bazelbuild/bazelisk/platforms"
@@ -51,12 +52,19 @@ func (gh *GitHubRepo) getFilteredVersions(bazeliskHome, bazelFork string, wantPr
 		return json.Marshal(releases)
 	}
 
-	url := fmt.Sprintf("https://api.github.com/repos/%s/bazel/releases", bazelFork)
+	forkSegments := strings.Split(bazelFork, "/")
+	if len(forkSegments) == 1 {
+		bazelFork = fmt.Sprintf("%s/bazel", bazelFork)
+	} else if len(forkSegments) > 2 {
+		return []string{}, fmt.Errorf("invalid bazel fork: %v", bazelFork)
+	}
+
+	url := fmt.Sprintf("https://api.github.com/repos/%s/releases", bazelFork)
 	auth := ""
 	if gh.token != "" {
 		auth = fmt.Sprintf("token %s", gh.token)
 	}
-	releasesJSON, err := httputil.MaybeDownload(bazeliskHome, url, bazelFork+"-releases.json", "list of Bazel releases from github.com/"+bazelFork, auth, merger)
+	releasesJSON, err := httputil.MaybeDownload(bazeliskHome, url, strings.Replace(bazelFork, "/", "-", -1)+"-releases.json", "list of Bazel releases from github.com/"+bazelFork, auth, merger)
 	if err != nil {
 		return []string{}, fmt.Errorf("unable to determine '%s' releases: %v", bazelFork, err)
 	}
