diff --git a/config/config.go b/config/config.go
index 4a22f51..98ef84d 100644
--- a/config/config.go
+++ b/config/config.go
@@ -128,6 +128,7 @@ type MappedKind struct {
 
 const ASPECT_WALKSUBDIR = "__aspect:walksubdir"
 const ASPECT_GITIGNORE = "__aspect:gitignore"
+const ASPECT_DIR_ENTRIES = "__aspect:direntries"
 
 func New() *Config {
 	return &Config{
diff --git a/walk/walk.go b/walk/walk.go
index 1b71732..af37ae7 100644
--- a/walk/walk.go
+++ b/walk/walk.go
@@ -143,6 +143,17 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[string]bool, updateRels *UpdateFilter, trie *pathTrie, wf WalkFunc, rel string, updateParent bool) ([]string, bool) {
 	haveError := false
 
+	// PATCH(fs.DirEntry map) ---
+	entsMap := make(map[string]fs.DirEntry, len(trie.children))
+	for _, node := range trie.children {
+		entsMap[node.entry.Name()] = node.entry
+	}
+	for _, node := range trie.files {
+		entsMap[node.Name()] = node
+	}
+	c.Exts[config.ASPECT_DIR_ENTRIES] = entsMap
+	// END PATCH(fs.DirEntry map) ---
+
 	// Absolute path to the directory being visited
 	dir := filepath.Join(c.RepoRoot, rel)
 
