diff --git a/walk/walk.go b/walk/walk.go
index 5503ee7..b073404 100644
--- a/walk/walk.go
+++ b/walk/walk.go
@@ -140,6 +140,17 @@ func Walk(c *config.Config, cexts []config.Configurer, dirs []string, mode Mode,
 func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[string]bool, updateRels *UpdateFilter, trie *pathTrie, wf WalkFunc, rel string, updateParent bool) ([]string, bool) {
 	haveError := false
 
+	// PATCH(fs.DirEntry map) ---
+	entsMap := make(map[string]fs.DirEntry, len(trie.children))
+	for _, node := range trie.children {
+		entsMap[node.entry.Name()] = node.entry
+	}
+	for _, node := range trie.files {
+		entsMap[node.Name()] = node
+	}
+	c.Exts["__aspect:direntries"] = entsMap
+	// END PATCH(fs.DirEntry map) ---
+
 	// Absolute path to the directory being visited
 	dir := filepath.Join(c.RepoRoot, rel)
 
