diff --git a/walk/walk.go b/walk/walk.go
index b2097b4..2c2bc08 100644
--- a/walk/walk.go
+++ b/walk/walk.go
@@ -473,9 +473,21 @@ func (w *walker) visit(c *config.Config, rel string, updateParent bool) {
 		w.errs = append(w.errs, err)
 	}
 	hasBuildFileError := err != nil
+
 	wc := info.config
 	containedByParent := info.file == nil && wc.updateOnly
 
+	// PATCH(files list) ---
+	filesMap := make(map[string]bool)
+	for _, f := range info.regularFiles {
+		filesMap[f] = true
+	}
+	for _, d := range info.subdirs {
+		filesMap[d] = true
+	}
+	c.Exts["__aspect:files"] = filesMap
+	// END PATCH(files list) ---
+
 	// Configure the directory, if we haven't done so already.
 	_, alreadyConfigured := w.visits[rel]
 	if !containedByParent && !alreadyConfigured {
@@ -519,9 +531,17 @@ func (w *walker) visit(c *config.Config, rel string, updateParent bool) {
 				return
 			}
 			for _, f := range vi.regularFiles {
+				// PATCH(files list) ---
+				filesMap[path.Join(prefix, f)] = true
+				// END PATCH(files list) ---
+
 				regularFiles = append(regularFiles, path.Join(prefix, f))
 			}
 			for _, f := range vi.subdirs {
+				// PATCH(files list) ---
+				filesMap[path.Join(prefix, f)] = true
+				// END PATCH(files list) ---
+
 				subdirs = append(subdirs, path.Join(prefix, f))
 			}
 			for _, subdir := range vi.subdirs {
