diff --git a/toolchain/cc_toolchain_config.bzl b/toolchain/cc_toolchain_config.bzl
index 6dfbfbfae..97a90be1a 100644
--- a/toolchain/cc_toolchain_config.bzl
+++ b/toolchain/cc_toolchain_config.bzl
@@ -231,7 +231,9 @@ def cc_toolchain_config(
 
     # TODO: The command line formed on darwin does not work with llvm-ar.
     ar_binary = tools_path_prefix + "llvm-ar"
-    if host_os == "darwin":
+
+    # WARNING: Added by silo patches.  Disable libtool linker altogether, it creates problems for us. 
+    if host_os == "darwin" and False:
         # Bazel uses arg files for longer commands; some old macOS `libtool`
         # versions do not support this.
         #
