diff --git a/toolchain/BUILD.llvm_repo b/toolchain/BUILD.llvm_repo
index a09616e..95f9280 100644
--- a/toolchain/BUILD.llvm_repo
+++ b/toolchain/BUILD.llvm_repo
@@ -19,7 +19,7 @@ exports_files(glob([
     "bin/*",
     "lib/*",
     "include/*",
-]))
+], allow_empty = True))
 
 ## LLVM toolchain files
 
@@ -45,12 +45,12 @@ filegroup(
     srcs = glob([
         "include/**/c++/**",
         "lib/clang/*/include/**",
-    ]),
+    ], allow_empty = True),
 )
 
 filegroup(
     name = "bin",
-    srcs = glob(["bin/**"]),
+    srcs = glob(["bin/**"], allow_empty = True),
 )
 
 filegroup(
@@ -67,6 +67,7 @@ filegroup(
             "lib/libclang*.a",
             "lib/liblld*.a",
         ],
+        allow_empty = True,
     ),
     # Do not include the .dylib files in the linker sandbox because they will
     # not be available at runtime. Any library linked from the toolchain should
