diff --git a/python/private/bzlmod_enabled.bzl b/python/private/bzlmod_enabled.bzl
index 84839981..9577bfd0 100644
--- a/python/private/bzlmod_enabled.bzl
+++ b/python/private/bzlmod_enabled.bzl
@@ -15,4 +15,4 @@
 
 # When bzlmod is enabled, canonical repos names have @@ in them, while under
 # workspace builds, there is never a @@ in labels.
-BZLMOD_ENABLED = "@@" in str(Label("//:unused"))
+BZLMOD_ENABLED = False
