diff --git a/gazelle/python/kinds.go b/gazelle/python/kinds.go
index a9483372..7a0639ab 100644
--- a/gazelle/python/kinds.go
+++ b/gazelle/python/kinds.go
@@ -32,7 +32,8 @@ func (*Python) Kinds() map[string]rule.KindInfo {
 
 var pyKinds = map[string]rule.KindInfo{
 	pyBinaryKind: {
-		MatchAny: true,
+		MatchAny: false,
+		MatchAttrs: []string{"srcs"},
 		NonEmptyAttrs: map[string]bool{
 			"deps":    true,
 			"main":    true,
