--- math/fp448/BUILD.bazel
+++ math/fp448/BUILD.bazel
@@ -10,6 +10,7 @@ go_library(
         "fp_generic.go",
         "fp_noasm.go",
     ],
+    cgo = True,
     importpath = "github.com/cloudflare/circl/math/fp448",
     visibility = ["//visibility:public"],
     deps = [
@@ -28,6 +29,8 @@ alias(
     visibility = ["//visibility:public"],
 )

+exports_files(["fp_amd64.h"], ["//visibility:public"])
+
 go_test(
     name = "fp448_test",
     srcs = [